components/ILIAS/Authentication/classes/LoginPage/LoginPageLanguagesOverviewTable.php

```diff
@@ -0,0 +1,226 @@
+            $query_params_namespace,
+            'action',
+            'key'
+        );
+
+        return [
+            'edit' => $this->ui_factory->table()->action()->single(
+                $this->lng->txt('edit'),
+                $overview_url_builder->withParameter($overview_action_parameter, 'edit'),
+                $overview_row_id
+            ),
+            'activate' => $this->ui_factory->table()->action()->standard(
+                $this->lng->txt('login_page_activate'),
+                $overview_url_builder->withParameter($overview_action_parameter, 'activate'),
+                $overview_row_id
+            ),
+            'deactivate' => $this->ui_factory->table()->action()->standard(
+                $this->lng->txt('login_page_deactivate'),
+                $overview_url_builder->withParameter($overview_action_parameter, 'deactivate'),
+                $overview_row_id
+            )
+        ];
+    }
+
+    private function initRecords(): void
+    {
+        if ($this->records === null) {
+            $this->records = [];
+            $i = 0;
+            $entries = $this->lng->getInstalledLanguages();
+            foreach ($entries as $langkey) {
+                $this->records[$i]['key'] = $langkey;
+                $this->records[$i]['id'] = ilLanguage::lookupId($langkey);
+                $status = ilAuthLoginPageEditorSettings::getInstance()->isIliasEditorEnabled(
+                    $langkey
+                );
+
+                $this->records[$i]['status_icon'] = $this->getStatusIcon($status);
+                $this->records[$i]['status'] = $status;
+                $this->records[$i]['language'] = $this->lng->txt('meta_l_' . $langkey);
+
+                ++$i;
+            }
+        }
+    }
+
+    private function getStatusIcon(bool $status): string
+    {
+        if ($status) {
+            $icon = $this->ui_renderer->render(
+                $this->ui_factory->symbol()->icon()->custom(
+                    \ilUtil::getImagePath('standard/icon_ok.svg'),
+                    $this->lng->txt('active')
+                )
+            );
+        } else {
+            $icon = $this->ui_renderer->render(
+                $this->ui_factory->symbol()->icon()->custom(
+                    \ilUtil::getImagePath('standard/icon_not_ok.svg'),
+                    $this->lng->txt('inactive')
+                )
+            );
+        }
+
+        return $icon;
+    }
+
+    public function getRows(
+        UI\Component\Table\DataRowBuilder $row_builder,
+        array $visible_column_ids,
+        Data\Range $range,
+        Data\Order $order,
+        ?array $filter_data,
+        ?array $additional_parameters
+    ): \Generator {
+        $records = $this->getRecords($range, $order);
+
+        foreach ($records as $record) {
+            $row_id = (string) $record['key'];
+            $deactivate_action = (bool) $record['status'] == true ? 'activate' : 'deactivate';
```

Comment:
Please move `activate` and `deactivate` to a private class constant.

Reply:
ok